org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/commands/CheckBoundGrammarCommand.java

```diff
@@ -0,0 +1,78 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.contentmodel.commands;
+
+import static org.eclipse.lemminx.extensions.xsd.utils.XSDUtils.TARGET_NAMESPACE_ATTR;
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.text.MessageFormat;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.extensions.contentmodel.participants.codeactions.NoGrammarConstraintsCodeAction;
+import org.eclipse.lemminx.services.IXMLDocumentProvider;
+import org.eclipse.lemminx.services.extensions.commands.AbstractDOMDocumentCommandHandler;
+import org.eclipse.lemminx.services.extensions.commands.ArgumentsUtils;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.utils.DOMUtils;
+import org.eclipse.lemminx.utils.FilesUtils;
+import org.eclipse.lemminx.utils.StringUtils;
+import org.eclipse.lsp4j.ExecuteCommandParams;
+import org.eclipse.lsp4j.jsonrpc.CancelChecker;
+import org.xml.sax.Attributes;
+import org.xml.sax.SAXException;
+import org.xml.sax.helpers.DefaultHandler;
+
+/**
+ * XML Command "xml.open.binding.wizard" to open the binding wizard for a
+ * DOM document from the command palette.
+ *
+ */
+public class CheckBoundGrammarCommand extends AbstractDOMDocumentCommandHandler {
+
+	public static final String COMMAND_ID = "xml.check.bound.grammar";
+
+	public CheckBoundGrammarCommand(IXMLDocumentProvider documentProvider) {
+		super(documentProvider);
+	}
+
+	@SuppressWarnings({ "serial" })
+	public static class AlreadyBoundException extends Exception {
+
+		private final static String MESSAGE = "Document already has bound grammar/schema";
+
+		public AlreadyBoundException() {
+			super(MESSAGE);
+		}
+
+	}
+
+	@Override
+	protected Object executeCommand(DOMDocument document, ExecuteCommandParams params, SharedSettings sharedSettings,
+			CancelChecker cancelChecker) throws Exception {
+		// Check if the document has a bound grammar.
+		try {
+			boolean shouldShow = !document.hasGrammar();
+			if (shouldShow) {
```

Comment:
Im sorry i have not explained correctly. This exception should be throwing in associategrammarcommand when user want to bind a xsd with a xml although it is already bound. Your check command should just return a booléen like you did before